harness/python/feature.py

```diff
@@ -121,3 +126,38 @@ async def check_result(self, handle: WorkflowHandle) -> None:
                     raise TypeError("Unexpected activity error") from err
             else:
                 raise err
+
+    async def get_history_events(self, handle: WorkflowHandle) -> list[HistoryEvent]:
+        next_page_token = b''
+        history: list[HistoryEvent] = []
+        request = GetWorkflowExecutionHistoryRequest()
+        request.namespace = self.namespace
+        request.execution.workflow_id = handle.id
+
+        while True:
+            request.next_page_token = next_page_token
+            response = await self.client.service.get_workflow_execution_history(request)
+            history.extend(response.history.events)
+            next_page_token = response.next_page_token
+            if not next_page_token:
+                break
+        return history
+
+
+class JSONPayload(TypedDict):
+    """
+    JSON proto Payload representation.
+    data and metadata values are base64 encoded
+    """
+    metadata: Mapping[str, str]
+    data: str
+
+
+def to_json_payload(payload: Payload) -> JSONPayload:
```

Comment:
Might make sense to just reuse the proto type here

Reply:
The initial TS implementation made sense to use this deserialized json for comparison but seems like the opposite in Go, Python works either way.
I'll find the best fit for all langs and refactor.